This is a small modification that changes how new tasks are handled.
Ordinarily, new tasks appear at the bottom of the task list. In this
view, you need to tap again to enter details like a due date.  With
this patch, when you add a task, a details card appears, in which you
can mmediately set the task name, due date, task list, and priority.
You're not forced to fill out each of those fields, they're just
available. Enter your info, then swipe back and you'll see your new
task in the list, or close the new task card (flick up) and the task
is saved automatically.

Mod by Tuckmobile
Patch submitted by JackieRipper
Index: /usr/palm/applications/com.palm.app.tasks/app/controllers/tasks-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.tasks/app/controllers/tasks-assistant.js
+++ /usr/palm/applications/com.palm.app.tasks/app/controllers/tasks-assistant.js
@@ -914,7 +914,7 @@ var TasksAssistant = Class.create({
 			this.listElement.mojo.insertItemsAt(index, [task]);
 			this.listElement.mojo.revealItem(index);
 			this.addWidgets(this.listElement.mojo.getNodeByIndex(index).querySelector('[name="subject"]'), task);
-			this.listElement.mojo.focusItem(task);
+			this.state.edit.call(this, index);
 			
 			task.priority = previousTask.priority; // Attempt to inherit the priority
 			
@@ -937,7 +937,7 @@ var TasksAssistant = Class.create({
 				{
 					// If we find the node we just added, focus on it.  We might not find the node however if the list is being refreshed
 					self.addWidgets(node.querySelector('[name="subject"]'), task);
-					self.listElement.mojo.focusItem(task);
+					self.state.edit.call(self, index);
 				}
 				
 				Mojo.Log.info("New task %j %j", task, self.task);
